tests/driver_at86rf2xx/main.c

```diff
@@ -93,11 +93,13 @@ int main(void)
 
     for (unsigned i = 0; i < AT86RF2XX_NUM; i++) {
         const at86rf2xx_params_t *p = &at86rf2xx_params[i];
+        netdev2_t *dev = (netdev2_t *)(&devs[i]);
+
         printf("Initializing AT86RF2xx radio at SPI_%d\n", p->spi);
         at86rf2xx_setup(&devs[i], p->spi, p->spi_speed, p->cs_pin,
                         p->int_pin, p->sleep_pin, p->reset_pin);
-        devs[i].driver->init((netdev2_t *)&devs[i]);
-        devs[i].event_callback = _event_cb;
+        dev->driver->init(dev);
```

Comment:
you must change the order here, once you call init, you might encounter external interrupts that might want to call _event_cb, and hard fault if that callback is not set, yet...